src/SixLabors.Fonts/Tables/Cff/Cff1Parser.cs

```diff
@@ -0,0 +1,1256 @@
+            FontDict[] fontDicts,
+            CffPrivateDictionary? privateDictionary)
+        {
+            // 14. CharStrings INDEX
+
+            // This contains the charstrings of all the glyphs in a font stored in
+            // an INDEX structure.
+
+            // Charstring objects contained within this
+            // INDEX are accessed by GID.
+
+            // The first charstring(GID 0) must be
+            // the.notdef glyph.
+
+            // The number of glyphs available in a font may
+            // be determined from the count field in the INDEX.
+
+            //
+
+            // The format of the charstring data, and therefore the method of
+            // interpretation, is specified by the
+            // CharstringType  operator in the Top DICT.
+
+            // The CharstringType operator has a default value
+            // of 2 indicating the Type 2 charstring format which was designed
+            // in conjunction with CFF.
+
+            // Type 1 charstrings are documented in
+            // the “Adobe Type 1 Font Format” published by Addison - Wesley.
+
+            // Type 2 charstrings are described in Adobe Technical Note #5177:
+            // “Type 2 Charstring Format.” Other charstring types may also be
+            // supported by this method.
+            reader.BaseStream.Position = this.offset + this.charStringsOffset;
+            if (!this.TryReadIndexDataOffsets(reader, out CffIndexOffset[]? offsets))
+            {
+                throw new InvalidFontFileException("No glyph data found.");
+            }
+
+#if DEBUG
+            if (offsets.Length > ushort.MaxValue)
+            {
+                throw new NotSupportedException();
+            }
+#endif
+            int glyphCount = offsets.Length;
+            var glyphs = new CffGlyphData[glyphCount];
+            Type2CharStringParser type2Parser = new(globalSubrRawBuffers, privateDictionary);
+
+#if DEBUG
+            double total = 0;
+#endif
+
+            // cid font or not
+            var fdRangeProvider = new FDRangeProvider(topDictionary.CidFontInfo.FdRanges);
+            bool isCidFont = topDictionary.CidFontInfo.FdRanges.Length > 0;
+
+            for (int i = 0; i < glyphCount; ++i)
+            {
+                CffIndexOffset offset = offsets[i];
+                byte[] buffer = reader.ReadBytes(offset.Length);
+#if DEBUG
+                // check
+                byte lastByte = buffer[offset.Length - 1];
+                if (lastByte is not (byte)Type2Operator1.Endchar and
+                    not (byte)Type2Operator1.Callgsubr and
+                    not (byte)Type2Operator1.Callsubr)
+                {
+                    // 5177.Type2
+                    // Note 6 The charstring itself may end with a call(g)subr; the subroutine must
+                    // then end with an endchar operator
+                    // endchar
+                    throw new Exception("invalid end byte?");
```

Comment:
again why only throw in Debug mode? If not only in debug mode, maybe a more specific exception/exception message .